File: 41-50/Q45.py
#Q45.py
class Solution:
    def jump(self, a):
        jumps = 0
        curEnd = 0
        far = 0
        for i in range(len(a)-1):
            far = max(far, i + a[i])
            if i == curEnd:
                jumps += 1
                curEnd = far
        return jumps

File: 41-50/Q45.cpp
//Q45.cpp
class Solution {
public:
    int jump(vector<int>& a) {
        int jumps = 0, curEnd = 0, far = 0;
        for(int i = 0; i < a.size()-1; i++){
            far = max(far, i + a[i]);
            if(i == curEnd){
                jumps++;
                curEnd = far;
            }
        }
        return jumps;
    }
};
